ethstore/src/dir/disk.rs

```diff
@@ -89,24 +90,32 @@ impl KeyDirectory for DiskDirectory {
 		Ok(accounts)
 	}
 
-	fn insert(&self, account: SafeAccount) -> Result<(), Error> {
+	fn insert(&self, account: SafeAccount) -> Result<SafeAccount, Error> {
 		// transform account into key file
-		let keyfile: json::KeyFile = account.into();
+		let keyfile: json::KeyFile = account.clone().into();
 
 		// build file path
-		let mut keyfile_path = self.path.clone();
-		keyfile_path.push(format!("{}", keyfile.id));
-
-		// save the file
-		let mut file = try!(fs::File::create(&keyfile_path));
-		try!(keyfile.write(&mut file).map_err(|e| Error::Custom(format!("{:?}", e))));
-
-		if let Err(_) = restrict_permissions_to_owner(&keyfile_path) {
-			fs::remove_file(&keyfile_path).expect("Expected to remove recently created file");
-			return Err(Error::Io(io::Error::last_os_error()));
+		let mut account = account;
```

Comment:
i don't see a cleaner way without making a major upheaval of the surrounding code.